crates/exex/exex/src/dyn_context.rs

```diff
@@ -0,0 +1,56 @@
+//! Mirrored version of [`ExExContext`](`crate::ExExContext`)
+//! without generic abstraction over [<Node>](`reth_node_api::FullNodeComponents`)
+
+use std::sync::Arc;
+
+use reth_chainspec::{EthChainSpec, Head};
+use reth_node_api::FullNodeComponents;
+use reth_node_core::node_config::NodeConfig;
+use tokio::sync::mpsc;
+
+use crate::{ExExContext, ExExEvent};
+
+// TODO(0xurb) - add `node` and `notifications` after abstractions
+/// Captures the context that an `ExEx` has access to.
+#[derive(Debug)]
+pub struct ExExContextDyn {
+    /// The current head of the blockchain at launch.
+    pub head: Head,
+    /// The config of the node
+    pub config: NodeConfig<Box<dyn EthChainSpec + 'static>>,
+    /// The loaded node config
+    pub reth_config: reth_config::Config,
+    /// Channel used to send [`ExExEvent`]s to the rest of the node.
+    ///
+    /// # Important
+    ///
+    /// The exex should emit a `FinishedHeight` whenever a processed block is safe to prune.
+    /// Additionally, the exex can pre-emptively emit a `FinishedHeight` event to specify what
+    /// blocks to receive notifications for.
+    pub events: mpsc::UnboundedSender<ExExEvent>,
+}
+
+impl<Node: FullNodeComponents> From<ExExContext<Node>> for ExExContextDyn {
```

Comment:
I dislike these from conversions because they make it less obvious that this is possible.

we can keep this from but this should always be paired with a function on `ExExContext` like `into_dyn_chainspec` or similar

Reply:
@shekhirin let's pair now with
```rust
	pub fn into_dyn_context(self) -> ExExContextDyn {
		ExExContextDyn::from(self)
	}
```
?

Reply:
@0xurb let's introduce an `into_dyn` method instead, I think we always want to make it explicit that you're converting to a struct with boxed traits